lib/galaxy/tools/evaluation.py

```diff
@@ -249,9 +249,8 @@ def wrap_input(input_values, input):
                 )
                 input_values[input.name] = wrapper
             elif isinstance(input, FieldTypeToolParameter):
-                if value is None:
-                    field_wrapper = None
-                else:
+                field_wrapper: Optional[Union[InputValueWrapper, DatasetFilenameWrapper]] = None
```

Comment:
I would remove the `field_wrapper` definition from here and move it to the first usage after ` if src == "json"` but without the `Optional`

``` python
if src == "json":
                        field_wrapper: Union[InputValueWrapper, DatasetFilenameWrapper] = InputValueWrapper(input, value, param_dict)
```

as the exception afterwards guarantees that `field_wrapper` will never be `None`

Reply:
I don't think we can remove the definition because `field_wrapper = None` is still used on line 266 (`input_values[input.name] = field_wrapper`) if `value` is `None`.